x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ProactiveStorageDeciderServiceTests.java

```diff
@@ -309,6 +311,41 @@ public void testForecast() {
         }
     }
 
+    public void testForecasWithTsdbDataStreamDoesNotFail() {
```

Comment:
This tests fails on 8.6 branch.

I also have a java rest integration test, which also fails on 8.6 branch. This test had to be modified due to changes between 8.6 and main branch. But the integration tests is more expensive to run. So I think just adding this unit test is ok? At least we have something that runs the forecast logic with tsdb data streams.